### PR TITLE
feature/US001-view-job-roles

### PR DESCRIPTION
**Pull request by japilarski**

### CODE REVIEW

File: agile-project-TEAMB.git/packed-refs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# pack-refs with: peeled fully-peeled sorted
```
Comment (ljercha) [inline agile-project-TEAMB.git/packed-refs]: What is that 👀 
I think you should get rid of that fork connection and clone repo to a separated one.

**Review by ljercha — CHANGES_REQUESTED** [review]

File: product-backend/scripts/US001.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+DROP DATABASE docker_JakubP;
+CREATE DATABASE docker_JakubP;
+use docker_JakubP;
+
+create table JobRoles (
+	jobRoleId INT primary key AUTO_INCREMENT,
```
Comment (leszekgu) [inline product-backend/scripts/US001.sql]: For sql columns please use snake_case https://en.wikipedia.org/wiki/Naming_convention_(programming)
Also it looks like there is a whitespace issue
Plus please align to structure from P002

File: product-backend/src/main/java/org/kainos/ea/resources/JobRoleController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.kainos.ea.api.JobRoleService;
+import org.kainos.ea.client.FailedToGetJobRolesException;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Api("Employees API")
```
Comment (leszekgu) [inline product-backend/src/main/java/org/kainos/ea/resources/JobRoleController.java]: This whole file still refers to employees in multiple places

File: product-web-app/service/jobRolesService.test.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+describe('JobRoleService', function () {
+    describe('getAllJobRoles', function () {
+        it('should return all job roles from response', async () => {
+            var mock = new MockAdapter(axios);
+            const data = [jobRole]
+            mock.onGet(jobRolesService['API_URL'] + '/job-roles').reply(200, data)
+            var results = await jobRolesService.getAllJobRoles();
+            // console.log('________________________________________')
```
Comment (leszekgu) [inline product-web-app/service/jobRolesService.test.ts]: comments can be removed if functionality is finished here

File: product-web-app/service/jobRolesService.test.ts
Comment (leszekgu) [inline product-web-app/service/jobRolesService.test.ts:1]: You are missing a case from acceptance criteria of having an empty list returned by API

File: product-web-app/service/jobRolesService.test.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            // console.log('jobRole value is: ', jobRole)
+            // console.log('result: ', results[0])
+            expect(results[0]).to.deep.equal(jobRole)
+        })
+        
+        it('should throw exception when 500 error returned from axios', async () => {
+            var mock = new MockAdapter(axios);
+            mock.onGet(jobRolesService['API_URL'] + 'job-roles').reply(500);
```
Comment (leszekgu) [inline product-web-app/service/jobRolesService.test.ts]: Is this really working? it differs  from line 19 on concatenation

File: product-backend/src/main/java/org/kainos/ea/api/JobRoleService.java
Comment (leszekgu) [inline product-backend/src/main/java/org/kainos/ea/api/JobRoleService.java:1]: I would like to see UnitTests for this class, happy path, exception and empty list coming from db

**Review by leszekgu — CHANGES_REQUESTED** [review]
whole directory 'agile-project-TEAMB.git' should be added to .gitignore and not part of this commit

File: .gitignore [hunk 6]
```diff
@@ -24,4 +24,6 @@ Icon
 .AppleDesktop
 Network Trash Folder
 Temporary Items
-.apdisk
\ No newline at end of file
+.apdisk
+
+agile-project-TEAMB.git
```
Comment (ljercha) [inline .gitignore:29]: is it needed?

File: product-backend/scripts/US001_create_db.sql [hunk 7]
```diff
@@ -0,0 +1,11 @@
+DROP DATABASE IF EXISTS TeamA_JakubP;
+CREATE DATABASE TeamA_JakubP;
```
Comment (ljercha) [inline product-backend/scripts/US001_create_db.sql]: schema name should be generic

File: product-backend/scripts/US001_example_data.sql [hunk 8]
```diff
@@ -0,0 +1,6 @@
+insert into JobRoles(job_role_title) values
```
Comment (ljercha) [inline product-backend/scripts/US001_example_data.sql]: scripts should not have the ticket number, use other ordering/counting way.

File: product-backend/src/main/java/org/kainos/ea/DropwizardWebServiceApplication.java [hunk 9]
```diff
@@ -31,7 +32,8 @@ protected SwaggerBundleConfiguration getSwaggerBundleConfiguration(DropwizardWeb
     @Override
     public void run(final DropwizardWebServiceConfiguration configuration,
                     final Environment environment) {
-        // TODO: implement application
+        environment.jersey().register(RolesAllowedDynamicFeature.class);
```
Comment (ljercha) [inline product-backend/src/main/java/org/kainos/ea/DropwizardWebServiceApplication.java:35]: what is that
```suggestion
```

File: product-backend/src/main/java/org/kainos/ea/resources/JobRoleController.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Api("API")
+@Path("/api")
+public class JobRoleController {
+    private JobRoleDao jobRoleDao = new JobRoleDao();
```
Comment (ljercha) [inline product-backend/src/main/java/org/kainos/ea/resources/JobRoleController.java]: not used
```suggestion
```

File: product-web-app/controller/JobRolesController.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import JobRoleService from "../service/jobRolesService.js";
+
+
+export default class JobRolesController {
+    constructor(app: Application) {
+        app.get("/job-roles", async (req: Request, res: Response) => {
+        const jobRoleService = new JobRoleService();
+        try {
+            let jobRoles: JobRole[] = await jobRoleService.getAllJobRoles();
+            // if (jobRoles.length=0) {
```
Comment (ljercha) [inline product-web-app/controller/JobRolesController.ts]: remove comments

File: product-web-app/service/jobRolesService.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import axios from "axios";
+import { JobRole } from "../model/jobRole.js";
+
+
```
Comment (ljercha) [inline product-web-app/service/jobRolesService.ts]: run ESLint to fix formatting, empty lines etc.
```
npm run lint
```

**Review by ljercha — CHANGES_REQUESTED** [review]

File: product-web-app/package.json [hunk 13]
```diff
@@ -13,6 +13,7 @@
   },
   "dependencies": {
     "axios": "1.4.0",
+    "axios-mock-adapter": "^1.21.5",
```
Comment (ljercha) [inline product-web-app/package.json]: Should be in devDependencies (run npm install after and commit package lock)

File: product-backend/scripts/US001_create_db.sql (hunk 7 above)
Comment (ljercha) [reply product-backend/scripts/US001_create_db.sql]: Did you apply the fix/convention from P002?

**Review by ljercha — COMMENTED** [review]
Few minors looks promising

File: product-backend/src/main/java/org/kainos/ea/resources/JobRoleController.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    private JobRoleService employeeService = new JobRoleService(jobRoleDao);
+
+    @GET
+    @Path("/job-roles")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getAllEmployees() {
```
Comment (leszekgu) [inline product-backend/src/main/java/org/kainos/ea/resources/JobRoleController.java]: why do you want to getEmployees? also employee is used in few other places in this class

**Review by leszekgu — CHANGES_REQUESTED** [review]
Looks better apply this one more change from me and some from Łukasz and soon it will go to testing

File: product-web-app/package.json (hunk 13 above)
Comment (ljercha) [reply product-web-app/package.json]: @japilarski  move it, run npm install and commit package-lock.json

**Review by ljercha — APPROVED** [review]

**Review by IamGalexing — APPROVED** [review]

File: product-backend/scripts/create_db.sql [hunk 15]
```diff
@@ -1,9 +1,11 @@
...
 
--- US001
+-- US001: view job roles
 create table JobRoles (
-    job_role_id INT primary key AUTO_INCREMENT,
+	job_role_id INT primary key AUTO_INCREMENT,
     job_role_title varchar(100)
 );
+
+select * from JobRoles;
```
Comment (leszekgu) [inline product-backend/scripts/create_db.sql]: that's not needed

Comment (japilarski) [reply product-backend/scripts/create_db.sql]: fixed

**Review by leszekgu — APPROVED** [review]

**Pull request merged**